### ISSUE TITLE
Events cell:mouseover and cell:mouseout work incorrectly

### ISSUE THREAD

**Issue by 40rn05lyv** [clientIO/joint#343]
Try to run following code in your browser.

```
var graph = new joint.dia.Graph;
var paper = new joint.dia.Paper({
  el: $('#paper'),
  width: 1600,
  height: 600,
  model: graph,
  gridSize: 1
});
var cells = [];
for (var i = 0; i < 10; i++) {
  var cell = new joint.shapes.basic.Rect({
    position : {
      x : i*120 + 100,
      y : 0
    },
    size: {
      width: 100,
      height: 100
    },
    attrs: {
      text: {
        text: 'This is text.'
      }
    }
  });
  cells.push(cell);
}
graph.addCell(cells);
paper.on("cell:mouseout", function(cellView) {
  alert("mouseout" + cellView.model.id);
})

```

If you try to place a cursor over text, you'll see an alert with "mouseout" message, despite the fact that you didn't leave this cell !

I've debugged this and it seems that backbone uses _jQuery.on(eventType, ".selector", func)_ in _delegate_ function.

Expected result: `cellMouseover` and `cellMouseout` in JointJs must ignore such events and be activated only when root `<g>` element of this cell activated the event.

**Comment by 40rn05lyv** [comment]
Actually, mouseover and mouseout works ok. We just need to have mouseenter/mouseleave events.

My code is following:

```
var CustomPaper = joint.dia.Paper.extend({
  events: function() {
    return _.extend({}, joint.dia.Paper.prototype.events, {
      'mouseenter .element': 'cellMouseenter',
      'mouseenter .link': 'cellMouseenter',
      'mouseleave .element': 'cellMouseleave',
      'mouseleave .link': 'cellMouseleave',
    });
  },
  cellMouseenter: function(evt) {
    evt = joint.util.normalizeEvent(evt);
    var view = this.findView(evt.target);
    if (view) {
      if (this.guard(evt, view)) return;
      view.mouseenter(evt);
    }
  },
  cellMouseleave: function(evt) {
    evt = joint.util.normalizeEvent(evt);
    var view = this.findView(evt.target);
    if (view) {
      if (this.guard(evt, view)) return;
      view.mouseleave(evt);
    }
  }
})

joint.dia.CellView.prototype.mouseenter = function(evt) {
  this.notify('cell:mouseenter', evt);
}
joint.dia.CellView.prototype.mouseleave = function(evt) {
  this.notify('cell:mouseleave', evt);
}
```

Now, we can create CustomPaper and set listeners:

```
var paper = new CustomPaper({ ... });
paper.on("cell:mouseenter", function(cellView) {
  alert("mouse entered!")
})
paper.on("cell:mouseleave", function(cellView) {
  alert("mouse left!")
})
```

**Comment by 40rn05lyv** [comment]
I think it worth adding this to the next release.

**Comment by kumilingus** [comment]
Hi, would you mind create a PR for this?

Sorry for late responses to your issues. We are currently incredibly busy with Rappid v2.0 release and other projects. Cheers.

**Comment by 40rn05lyv** [comment]
What is PR?

**Comment by kumilingus** [comment]
A pull request.

**Comment by kumilingus** [comment]
Implemented in #442. Thank you for the suggestion.
